### DISCUSSION TITLE
How to use HandyControl in a wpf library.

### DISCUSSION THREAD

**Discussion by saedrna** [HandyOrg/HandyControl#843] [Q&A] [answered]
I am now developing a plugin for other applications. I can only inject library (dll) into the app. Therefore, I have no access to ```App.xaml``` or ```Application.Resources``. 
I am now putting all the HandyControl resources inside something like 
```
    <UserControl.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <hc:ThemeResources/>
                <hc:Theme/>
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </UserControl.Resources>
```

However, an annoying problem is that, it seems some style is lost after create the window twice, e.g. close and reopen an window, like below.
![image](https://user-images.githubusercontent.com/3850191/119967132-9c3fbd80-bfde-11eb-8fd5-a89c45f657ec.png)

Is there a way to solve this?
Thanks,
Han

**Comment by NaBian** [comment, marked as answer]
#603

**Reply by saedrna** [reply]
我就是这样做的。找到原因了，因为我用的是非官方的 HandyControls。

**Reply by saedrna** [reply]
另外，我之所以换非官方的版本，是因为官方版 ```SelectionChanged``` 事件响应有问题。
```
<hc:SideMenu>
            <hc:Interaction.Triggers>
                <hc:EventTrigger EventName="SelectionChanged">
                    <hc:EventToCommand Command="{Binding CommandSwitchPage}" PassEventArgsToCommand="True"/>
                </hc:EventTrigger>
            </hc:Interaction.Triggers>
<hc:SideMenu>
```
这里 无法正常绑定 ```CommandSwitchPage```，我用的 ReactiveUI的 ReactiveCommand，同样代码非官方版本可以绑定。
```
       CommandSwitchPage = ReactiveCommand.Create<FunctionEventArgs<object>>(
                (x) => { SwitchPage(x); });

        public ReactiveCommand<FunctionEventArgs<object>, Unit> CommandSwitchPage { get; set; }

```

这个如何解决呢？

**Reply by NaBian** [reply]
这个问题不予以回答，请参考 demo
